core/controllers/reader_test.py

```diff
@@ -1488,50 +1447,71 @@ def test_playthrough_not_stored_at_limiting_value(self):
             '/explorehandler/store_playthrough/%s' % (self.exp_id),
             {
                 'playthrough_data': self.playthrough_data,
-                'exp_issue_dict': self.exp_issue_dict
+                'issue_schema_version': 1
             }, self.csrf_token)
         self.process_and_flush_pending_tasks()
 
         model = stats_models.ExplorationIssuesModel.get_model(self.exp_id, 1)
         self.assertEqual(len(model.unresolved_issues), 1)
         self.assertEqual(len(model.unresolved_issues[0]['playthrough_ids']), 5)
 
-    def test_error_on_invalid_exp_issue_dict(self):
-        """Test that passing an invalid exploration issue dict raises an
+    def test_error_without_schema_version_in_payload_dict(self):
+        """Test that passing a payload without schema version raises an
         exception.
         """
-        del self.exp_issue_dict['issue_type']
-
         self.post_json(
             '/explorehandler/store_playthrough/%s' % (self.exp_id),
             {
                 'playthrough_data': self.playthrough_data,
-                'exp_issue_dict': self.exp_issue_dict
             }, self.csrf_token, expect_errors=True, expected_status_int=400)
 
-    def test_error_without_schema_version_in_exp_issue_dict(self):
-        """Test that passing an exploration issue dict without schema version
-        raises an exception.
-        """
-        del self.exp_issue_dict['schema_version']
+    def test_error_on_invalid_playthrough_dict(self):
+        """Test that passing an invalid playthrough dict raises an exception."""
+        self.playthrough_data['issue_type'] = 'FakeIssueType'
 
         self.post_json(
             '/explorehandler/store_playthrough/%s' % (self.exp_id),
             {
                 'playthrough_data': self.playthrough_data,
-                'exp_issue_dict': self.exp_issue_dict
+                'issue_schema_version': 1
             }, self.csrf_token, expect_errors=True, expected_status_int=400)
 
-    def test_error_on_invalid_playthrough_dict(self):
-        """Test that passing an invalid playthrough dict raises an exception."""
-        self.playthrough_data['issue_type'] = 'FakeIssueType'
+    def test_playthrough_id_is_returned_and_subsequently_updated(self):
+        """Test that playthrough ID is returned when it is stored for the first
+        time and the playthrough is updated from the next time.
```

Comment:
Please move the "update" test to its own testcase (where you don't check validity of the first-stored playthrough).

Reply:
Done.